remote.go

```diff
@@ -33,15 +33,16 @@ func StartRemote(cfg *config.ServerConfig) {
 		if err != nil {
 			log.Fatal(err)
 		}
-	case config.TLS:
```

Comment:
My idea here is that since TLS is just TCP with a wrapper that we eliminate the TLS option. Just have TCP - and if there are TLS options set in the config then have TCP use those to enable TLS sessions. This idea mostly sprang out of the fact that for HTTP2 to work with TLS it has to have a special TLS handshake - so the original way we would have the top-level `remote/server.go` have a TLS listener and hand down a TLSConn wouldn't work for HTTP2. The gist is to now update the TCPHandler to read the config it's given and determine if it should use TLS or not